### PR TITLE
MGMT-18618 Fix local rhsso deployment of assisted-installer

### PR DESCRIPTION
**Pull request by dudyas6**
This PR adds the option to pass the JWKS_URL environment variable to the assisted-service when deploying with rhsso authentication. Additionally, it includes a fix for a minor logic issue in the Makefile where we create_hub_cluster before cloning assisted-service repo locally.

### CODE REVIEW

File: src/update_assisted_service_cm.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
     ("HW_VALIDATOR_MIN_CPU_CORES_SNO", ""),
     ("HW_VALIDATOR_MIN_RAM_GIB_SNO", ""),
     ("ENABLE_UPGRADE_AGENT", "true"),
+    ("JWKS_URL", ""),
```
Comment (danmanor) [inline src/update_assisted_service_cm.py:43]: This should be the default right ?
https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/certs

File: README.md [hunk 2]
```diff
@@ -128,6 +128,7 @@ before you run start the deployment.
 | `OCM_BASE_URL`                          | OCM API URL used to communicate with OCM and AMS, default: https://api.integration.openshift.com/ |
 | `OCM_CLIENT_ID`                         | ID of Service Account used to communicate with OCM and AMS for Agent Auth and Authz |
 | `OCM_CLIENT_SECRET`                     | Password of Service Account used to communicate with OCM and AMS for Agent Auth and Authz |
+| `JWKS_URL`                              | URL for retrieving the JSON Web Key Set (JWKS) used for verifying JWT tokens in authentication. Defaults to https://api.openshift.com/.well-known/jwks.json |
```
Comment (danmanor) [inline README.md:131]: this is not the default right ?

**Review by danmanor — CHANGES_REQUESTED** [review]

File: src/update_assisted_service_cm.py (hunk 1 above)
Comment (dudyas6) [reply src/update_assisted_service_cm.py:43]: It should be but i didn't want to change current configurations, only to add the option to insert it as env param, should change it to default?

File: README.md (hunk 2 above)
Comment (dudyas6) [reply README.md:131]: It is currently, should we change it to be the working param? didnt want to change current configurations as mentioned above

Comment (danmanor) [reply README.md:131]: yes looks like it doesn't work

File: src/update_assisted_service_cm.py (hunk 1 above)
Comment (danmanor) [reply src/update_assisted_service_cm.py:43]: yes looks like it doesn't work

**Comment by danmanor** [comment]
@dudyas6 ignore `ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node`

**Comment by danmanor** [comment]
/override ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node

**Review by danmanor — APPROVED** [review]

**Comment by danmanor** [comment]
/override ci/prow/e2e-metal-assisted-kube-api-late-binding-single-node

**Comment by danmanor** [comment]
overriding e2e-metal-assisted (the current changes only affect the deployment, the failures are conformance tests)

**Pull request merged**